dynamiq/nodes/llms/openai.py

```diff
@@ -21,3 +28,88 @@ def __init__(self, **kwargs):
         if kwargs.get("client") is None and kwargs.get("connection") is None:
             kwargs["connection"] = OpenAIConnection()
         super().__init__(**kwargs)
+
+    def is_o_series_model(self) -> bool:
+        """Determine if the model belongs to the O-series (e.g. o1 or o3).
+
+        Returns:
+            bool: True if the model is an O-series model, otherwise False.
+        """
+        model_lower = self.model.lower()
+        return "o1" in model_lower or "o3" in model_lower
```

Comment:
Please define some ClassVar constant tuple like `o_series_model_prefixes`. After go over this tuple and check if the model starts with this prefix (not just check for substring).